### DISCUSSION TITLE
table的cols中有字段为edit为函数写法时，怎么能导出列配置包括函数

### DISCUSSION THREAD

**Discussion by hjingsuper** [layui/layui#1559] [讨论]
![image](https://github.com/layui/layui/assets/36070786/4c9caf8b-3098-48f9-8344-16406a9df293)
![image](https://github.com/layui/layui/assets/36070786/458edb33-348e-4403-b57a-6a511d38aa11)

不同页面的table渲染方式不一致，有的是【方法配置渲染】，有的是【模板配置渲染】，其中还包括字段为edit为函数写法。此时想整合页面时，有什么方式能使cols保持一致？
1.使用JSON.stringify(options.cols)只能打印基础配置属性，无法得到函数

**Comment by sentsim** [comment]
尝试将 cols 的配置单独定义

```
var cols = [[…]];
```

1. 若 cols 定义在全局（且在 layui.js 之前），那么可在模板中直接设置 `cols: cols`
2. 若 cols 局部定义，可用 `table.init(id,  { cols: cols })` 给模板配置的实例重新初始化；至于 `table.render()` 的渲染方式，直接将定义的 cols 赋值到 cols 属性上即可。

**Reply by hjingsuper** [reply]
我的意思是，已有的页面的table有不同的渲染方式，每个页面的字段也是不同的。现在想将不同页面的字段汇集起来，使用JSON.stringify(options.cols)将不同页面的列都取出来，进行模板配置渲染，但是取不到edit的函数写法
以下是最小重现：将demo中table1的列加入到table中，有点不知道怎么处理
demo:[https://stackblitz.com/edit/web-platform-3r7bpa?file=index.html](https://stackblitz.com/edit/web-platform-3r7bpa?file=index.html)

**Reply by sentsim** [reply]
我真的不大理解为何要将 `cols` 系列化，或许这道题等 @Sight-wcg 来解 😅

**Comment by hjingsuper** [comment]
说白了，就是使JSON.stringify保留函数，使用以下code得到函数字符串解决😁
[https://www.cnblogs.com/yalong/p/15768087.html](https://www.cnblogs.com/yalong/p/15768087.html)
`
const stringify = (obj) => {
  return JSON.stringify(obj, (k, v) => {
    if(typeof v === 'function') {
        return `${v}`
    } else {
      return v
    }
  })
};console.log(stringify(options.cols));
`

**Comment by hjingsuper** [comment]
已解决
